### DISCUSSION TITLE
Minimal latency and unreliable streams

### DISCUSSION THREAD

**Discussion by okbaswc** [microsoft/msquic#1186] [Q&A] [answered]
How is it possible with MSQUIC to drop the reliability/order/acknowledgements of a stream, and get as close to raw UDP as possible in latency?

**Comment by nibanks** [comment, marked as answer]
Per-spec streams are reliable, ordered streams of data. You cannot just "drop" that feature, but there are two possible solutions to your problem:

1. You use Datagrams ([DatagramSend](https://github.com/microsoft/msquic/blob/main/docs/api/DatagramSend.md)) instead of streams.  This allows you to send and receive chunks (not streams) of data over the QUIC connection. There is less overhead than streams.
2. We _could_ add support for delivering stream data immediately to the app as we receive it, not buffering it or trying to reorder out of order data, but then the app would have to handle all that. I'm not sure this would help all that much though.
